proto/kuksa/val/v2/types.proto

```diff
@@ -147,18 +159,20 @@ message ValueRestrictionString {
   repeated string allowed_values = 1;
 }
 
+// Used to indicate status of a signal in Databroker
+// This is given as an alternative to Value, so if there is a valid value
+// ValueFailure shall not be specified.
```

Comment:
Just a thought - would it be better to name this enum `ValueInfo`? At least getting `NOT_PROVIDED` is not necessarily a failure, and could theoretically be used by some provider in `PublishValueRequest` to indicate that the previous value should be deleted.

Reply:
I agree that calling it failure is a bit misleading. I also think having the value inside a `value_state` is unnecessary as it adds one more level of nesting to get to the actual value.

I think something like this would be more intuitive:

```proto
message DataPoint {
  google.protobuf.Timestamp timestamp = 1;

  Value value = 3;

  Status status = 10;
}

enum Status {
  STATUS_UNSPECIFIED = 0;  
  STATUS_OK = 1;
  STATUS_VALUE_NOT_AVAILBLE = 2;
}
```

If there is no value, a provider (or databroker) can set it to `None` and the corresponding reason in the `status` field.

One "downside" of this is that it's possible to set a value while at the same time setting the status to something other than STATUS_OK (and vice versa setting STATUS_OK while not setting any value).

But doing that can simply be disallowed by databroker in the same way that it doesn't allow setting a value of the wrong data type today.

Reply:
I like the proposal.

And if we document (in *.proto) how various combinations/situations should be handled by client and databroker it should not be a problem.

| Scenario | value | status |
|---|---|---|
| Value available for signal | `<value>` | `STATUS_UNSPECIFIED` or `STATUS_OK`
| Value NOT available for signal | `None` | `STATUS_VALUE_NOT_AVAILABLE`

Anyway - will not have time for any update next week, but @rafaeling or @lukasmittag  are free to change if they like and gree

Reply:
Also agree with that, like this approach more than the current one.